cfme/physical/provider/redfish.py

```diff
@@ -0,0 +1,65 @@
+import attr
+
+from widgetastic_patternfly import BootstrapSelect, Input
+from wrapanapi.systems import RedfishSystem
+
+from cfme.common.provider import DefaultEndpoint, DefaultEndpointForm
+from . import PhysicalProvider
+
+
+class RedfishEndpoint(DefaultEndpoint):
+    api_port = 443
+    security_protocol = 'SSL'
+
+    @property
+    def view_value_mapping(self):
+        return {
+            'security_protocol': self.security_protocol,
+            'hostname': self.hostname,
+            'api_port': self.api_port
+        }
+
+
+class RedfishEndpointForm(DefaultEndpointForm):
+    security_protocol = BootstrapSelect('default_security_protocol')
+    api_port = Input('default_api_port')
+
+
+@attr.s(hash=False)
+class RedfishProvider(PhysicalProvider):
+    STATS_TO_MATCH = ['num_server']
+    type_name = 'redfish'
+    endpoints_form = RedfishEndpointForm
+    string_name = 'Physical Infrastructure'
+    mgmt_class = RedfishSystem
+    refresh_text = "Refresh Relationships and Power States"
+    db_types = ["Redfish::PhysicalInfraManager"]
+    settings_key = 'ems_redfish'
+    log_name = 'redfish'
+
+    @property
+    def mgmt(self):
+        from cfme.utils.providers import get_mgmt
+        d = self.data
```

Comment:
**question** is it ok for the provider/metadata that self.data is modified here

Reply:
`self.data` is actually a property and not a hash. Knowing this I assume it's safe.

Reply:
Oh I'm wrong, it is a property, but it returns internal hash. Sorry bout that, looks like `copy.deepcopy(self.data)` has to be here after all.